provider/service.go

```diff
@@ -41,7 +41,7 @@ type Service interface {
 
 // NewService creates and returns new Service instance
 // Simple wrapper around various services needed for running a provider.
-func NewService(ctx context.Context, session session.Session, bus pubsub.Bus, cclient cluster.Client) (Service, error) {
+func NewService(ctx context.Context, session session.Session, bus pubsub.Bus, cclient cluster.Client, clusterPublicHostname string, clusterExternalPortQuantity uint) (Service, error) {
```

Comment:
yeah let's populate a config object to pass in here instead of adding more and more params.